index.bs

```diff
@@ -1436,6 +1455,31 @@ To <dfn export>get user context</dfn> given |user context id|:
 
 </div>
 
+When a [=user context=] is [=set/remove|removed=] from the
+[=set of user contexts=], [=remove user context subscriptions=]:
+
+<div algorithm>
+
+To <dfn export>remove user context subscriptions</dfn>:
```

Comment:
So, this does create undefined behaviour, because it makes it observable whether a specific user context was removed or not, which is implementation defined (I guess technically that was already the case since it affects what's in storage). It isn't a blocker here, but I wonder if we should define that any user context created via `createUserContext` actually is removed if it goes from 1 to 0 browsing contexts.

Reply:
I am not sure I get why this creates undefined behavior. I added this algorithm to define a clean-up step to avoid accumulating subscriptions if the client does not unsubscribe (unsubscribe requests with the ID of removed subscriptions can still be issued). I think it is not observable to the local end. Could you please clarify?

>  It isn't a blocker here, but I wonder if we should define that any user context created via createUserContext actually is removed if it goes from 1 to 0 browsing contexts.

In Chrome, the user context stays there if all browsing contexts in it are removed.

Reply:
I guess it's actually not externally visible whether or not this cleanup happened, so it's fine.

The problem I'm thinking of is actually already an issue. Say you create a user context, create a tab in that user context, then close the tab. Now an implementation is permitted to remove the user context. But whether or not you do that is visible if you then try to create another new tab in that user context.

It kind of makes sense for gecko because you can delete the user context in the browser UI. So if someone actually interacts with the browser and does that all bets are off. But maybe the spec should be clearer that it's not expected that user contexts (or their associated storage) are removed without user interaction.

Reply:
So the same situation can happen in Chrome if the user closes a user context via the UI. I think it makes sense to clarify that user contexts should not get removed randomly by the implementation, but removing them as the result of a direct user action is okay.